pkgs/shells/zsh/zsh-fast-syntax-highlighting/default.nix

```
@@ -0,0 +1,41 @@
+{ stdenv, lib, fetchFromGitHub }:
+
+stdenv.mkDerivation rec {
+  pname = "zsh-fast-syntax-highlighting";
+  version = "2019-07-20";
+
+  src = fetchFromGitHub {
+    owner = "zdharma";
+    repo = "fast-syntax-highlighting";
+    rev = "ab78177ee23a6509be7e8fe09d265d5640fb333a";
+    sha256 = "1h65ydb90wzl0f2w0q3sgc77l1zdn6w33fv8fg5nvx2ycl2xr43x";
+  };
+
+  #adapted from https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=zsh-fast-syntax-highlighting-git
+  installPhase = ''
+    _plugindir="$out/share/zsh/site-functions"
+    _chromadir="$_plugindir/chroma"
+    _themesdir="$_plugindir/themes"
+
+    cd "${src}"
```

Please don't use string interpolation, use the environment variable instead to allow users to override it.
```suggestion
    cd "$src"
```
Also this installPhase is a bit too complicated for it's functionality (copy files from `"$src"` to `"$out"/share/zsh/site-functions`)...